db-service/lib/deep-queries.js

```diff
@@ -240,7 +240,7 @@ const _getDeepQueries = (diff, target) => {
       queries.push(INSERT.into(target).entries(diffEntry))
     } else if (op === 'delete') {
       queries.push(DELETE.from(target).where(diffEntry))
-    } else if (op === 'update' || (op === undefined && subQueries.length && _hasManagedElements(target))) {
+    } else if (op === 'update' || (op === undefined && _hasManagedElements(target))) {
```

Comment:
This doesn't just impact drafts. This check was extra added to only update the parent if children where changed, because there are tests which expect exactly that.

So we have to make sure whether those tests are having the wrong expectation. As it is now no longer allowed to differentiate between draft and non draft entities.

@David-Kunz @johannes-vogel could you provide some clarification on the expected behavior?

Reply:
> This check was extra added to only update the parent if children where changed, because there are tests which expect precisely that.

The expectation regarding the deep update scenario in the draft context was outlined in the PR and discussed within the team. In a draft-enabled deep update scenario, the parent entity must always be updated after an edit operation, regardless of any changes made to the children's data. This decision was taken long ago, and to my knowledge, it has always been the case when utilizing the default database layer.

So the question is whether the outlined behavior is also expected in a non-draft deep update scenario.

Reply:
> So the question is whether the outlined behavior is also expected in a non-draft deep update scenario.

Exactly

Reply:
@BobdenOs

> because there are tests which expect exactly that.

Could you point out which runtime integration tests expect the opposite behavior? Why are the tests not already failing?

The code in question was added in this PR without any tests:
https://github.com/cap-js/cds-dbs/pull/15

Reply:
Found the test that fails `cds/tests/_runtime/odata/__tests__/integration/etag.test.js` -> `composition to many`

Which expects that children with managed data that are not touched should not change.